weblab/experiments/tests/test_views.py

```diff
@@ -892,6 +897,93 @@ def test_view_experiment_version(self, client, experiment_version):
         assert response.context['version'] == experiment_version
 
 
+@pytest.mark.django_db
+class TestExperimentTasks:
+
+    @pytest.mark.usefixtures('other_user')
+    def test_load_page_other_user(self, client):
+        response = client.get('/experiments/tasks')
+        assert response.status_code == 302
+
+    @pytest.mark.usefixtures('logged_in_user')
+    def test_load_page_logged_in_user(self, client):
+        response = client.get('/experiments/tasks')
+        assert response.status_code == 200
+
+    @pytest.mark.usefixtures('logged_in_user')
+    def test_get_queryset_other_user(self, other_user, client, experiment_version):
+        experiment_version.author = other_user
+        experiment_version.save()
+        recipes.running_experiment.make(experiment_version=experiment_version)
+        assert RunningExperiment.objects.count() == 1
+        response = client.get('/experiments/tasks')
+        assert len(response.context['runningexperiment_list']) == 0
+
+    def test_get_queryset(self, logged_in_user, client, helpers):
+        # Create three experiment versions 2 running and 1 completed
+        model_1 = recipes.model.make(author=logged_in_user)
+        model_1_version = helpers.add_version(model_1, visibility='public')
+        protocol_1 = recipes.protocol.make(author=logged_in_user)
+        protocol_1_version = helpers.add_version(protocol_1, visibility='public')
+        protocol_1_version2 = helpers.add_version(protocol_1, visibility='public')
+        protocol_2 = recipes.protocol.make(author=logged_in_user)
+        protocol_2_version = helpers.add_version(protocol_2, visibility='public')
+
+        recipes.experiment_version.make(
+            status=ExperimentVersion.STATUS_SUCCESS,
+            experiment__model=model_1,
+            experiment__model_version=model_1_version.hexsha,
+            experiment__protocol=protocol_1,
+            experiment__protocol_version=protocol_1_version.hexsha,
+            author=logged_in_user,
+        )
+
+        exp_version_2 = recipes.experiment_version.make(
+            status=ExperimentVersion.STATUS_RUNNING,
```

Comment:
Actually, maybe make this one QUEUED to prove that both statuses count?

Reply:
good idea - done